### ISSUE TITLE
racket-xp-mode reports "defined locally" and imported, both?

### ISSUE THREAD

**Issue by greghendershott** [greghendershott/racket-mode#639]
In this little program:

```racket
#lang racket
(define-syntax when-not (make-rename-transformer #'unless))
(unless #f 42)
```

`racket-xp-mode` reports that `unless` is both "imported from `racket`", _and_, "defined locally":

![arrow-rename-transformer-emacs](https://user-images.githubusercontent.com/265119/187938156-55033a14-6558-416f-ad91-2f0284157fe7.png)

Racket Mode adds the `"defined locally"` when [`syncheck:add-arrow/name-dup/pxpy`](https://docs.racket-lang.org/drracket-tools/Accessing_Check_Syntax_Programmatically.html#%28meth._%28%28%28lib._drracket%2Fcheck-syntax..rkt%29._syncheck-annotations~3c~25~3e%29._syncheck~3aadd-arrow%2Fname-dup%2Fpxpy%29%29) has a false `require-arrow`.

---

DrRacket using arrows:

With the mouse pointer over `unless`:

![arrow-rename-transformer-drracket](https://user-images.githubusercontent.com/265119/187938318-de7d62a2-848b-4acf-8e78-d54d983a8b28.png)

With the mouse pointer over `when-not` (disregard the caret in the screenshot, it's misleading; the mouse pointer is actually over `when-not`):

![arrow-rename-transformer-drracket-2](https://user-images.githubusercontent.com/265119/187943509-6bd4de85-b909-42cd-8f7c-c33c5ae1200f.png)

In other words there is a non-import arrow from `unless` to `when-not` -- but also a non-import arrow in the opposite direction. This doesn't make sense to me; I'm not sure if it's a small bug in check-syntax, or, a big defect in my understanding of macros. Anyway, given that information, I don't see how Racket Mode can do otherwise here. Fortunately jump-to-definition does work correctly, going from `unless` to its actual definition (not to `when-not`).

**Comment by greghendershott** [comment]
While trying to revive a slightly stale project (that captures check-syntax runs into a database, to help support things like a project-wide "find references" command), I noticed some tests were failing, today, that I'm pretty sure did not fail about a year ago.

So I suspect that something changed in drracket/check-syntax within the last year or so, to make this happen (or at least, make it happen much more frequently). So I guess I'll need to detour into the commit history for that, and see what I can figure out.

**Comment by greghendershott** [comment]
In fact `actual?` is N/A.

I think the way to fix this is to inspect the set of annotations at the end, and if it contains both "defined locally" and #"^imported from", the delete the former. This is a bit kludgy, but it originates from us adding the "defined locally" annotation ourselves when `syncheck:add-arrow` is called with `require-arrow?` false. If it is also called another time (before or after) with `require-arrow` non-false, we should trust that. The `require-arrow?` false case can arise from a rename transformer; the false value is not intended as a reliable signal that it is non-imported binding.
